src/index.js

```diff
@@ -1,58 +1,128 @@
+          </svg> 
+      </div>
     </li>
-    `).join('');
+    `;
+
+  clearCompleted.insertAdjacentHTML('beforebegin', liMarkup);
+};
+
+const clearTaskInputs = () => {
+  const inputItem = document.querySelector('#input-item');
+  inputItem.value = '';
+};
+
+const deleteTask = (element) => {
+  element.remove();
+};
+
+const editTask = (index, value) => {
+  const tasks = JSON.parse(localStorage.getItem('tasks'));
+  tasks[index].description = value;
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+};
+
+// Local Storage Functions
+const getTasks = () => {
+  let tasks;
+  if (localStorage.getItem('tasks') === null) tasks = [];
+  else tasks = JSON.parse(localStorage.getItem('tasks'));
+
+  return tasks;
+};
 
-  taskInput.insertAdjacentHTML('afterend', liMarkup);
+const addTaskLocalStorage = (task) => {
+  const tasks = getTasks();
+  tasks.push(task);
+  localStorage.setItem('tasks', JSON.stringify(tasks));
 };
 
-window.document.addEventListener('DOMContentLoaded', contentMarkup);
+const deleteTaskLocalStorage = (element) => {
+  const tasks = getTasks();
+  tasks.forEach((task, index) => {
+    if (element === index) tasks.splice(index, 1);
+  });
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+};
+
+const updateIndex = () => {
+  window.location.reload();
+  const tasks = JSON.parse(localStorage.getItem('tasks'));
+
+  for (let i = 0; i < tasks.length; i += 1) {
+    tasks[i].index = i + 1;
+  }
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+};
+
+window.document.addEventListener('DOMContentLoaded', () => {
+  let indexV = 1;
+
+  const tasks = getTasks();
+  tasks.forEach((task) => {
+    addTask(task);
+    indexV += 1;
+  });
+
+  const addItem = document.querySelector('.add-item');
+  const inputItem = document.querySelector('#input-item');
+
+  addItem.addEventListener('click', () => {
+    if (inputItem.value !== '') {
+      const tasksObject = new Tasks(indexV, inputItem.value, false);
+      addTask(tasksObject);
+      indexV += 1;
+      updateIndex();
+      addTaskLocalStorage(tasksObject);
+      clearTaskInputs();
+    } else {
+      //
+    }
+  });
+
+  const taskDots = Array.from(document.getElementsByClassName('delete-task'));
+
+  taskDots.forEach((element, index) => {
+    element.addEventListener('click', (e) => {
+      const node = e.target.parentNode.parentNode;
+      deleteTask(node);
+      indexV -= 1;
+      deleteTaskLocalStorage(index);
+      updateIndex();
+    });
+
+    const parent = element.parentNode.parentNode;
+    const textAreaContent = parent.querySelector('textarea');
+
+    textAreaContent.addEventListener('blur', () => {
+      editTask(index, textAreaContent.value);
+    });
+  });
+});
```

Comment:
- [ ] **In your README File:** As it is required for all **MICROVERSE** project to have a descriptive README. Kindly make sure you update your current one filling all the concerned sections of your project withe expected informations(e.g.: **Description**, **Getting Started**, **Authors**,...) as you are still using **placeholders of the template**. This will make your README be more professional. To help you filling your sections I suggest you use **[this template](https://github.com/tresorsawasawa/Todo_List_App/blob/main/README.md)**

Reply:
@tresorsawasawa the **REAME File** has been updated